src/routes/relay/relay.controller.spec.ts

```diff
@@ -49,151 +65,298 @@ describe('RelayController', () => {
   });
 
   describe('/v1/relay (POST)', () => {
-    it('should return a 201 when the body is valid', async () => {
-      axios.default.get = jest.fn().mockResolvedValue({ data: 'mockSafe' });
-
-      const relayServiceSpy = jest
-        .spyOn(relayService, 'sponsoredCall')
-        .mockImplementation(() => Promise.resolve({ taskId: '123' }));
-
-      const body = {
-        chainId: '5',
-        to: faker.finance.ethereumAddress(),
-        data: MOCK_EXEC_TX_CALL_DATA,
-      };
-
-      await request(app.getHttpServer())
-        .post('/v1/relay')
-        .send(body)
-        .expect(201);
-
-      expect(relayServiceSpy).toHaveBeenCalledTimes(1);
+    describe('Relayer', () => {
+      it('should return a 201 when the body is valid', async () => {
+        axios.default.get = jest.fn().mockResolvedValue({ data: 'mockSafe' });
+
+        const sponsorServiceSpy = jest
+          .spyOn(mockSponsorService, 'sponsoredCall')
+          .mockImplementation(() => Promise.resolve({ taskId: '123' }));
+
+        const body = {
+          chainId: '5',
+          to: faker.finance.ethereumAddress(),
+          data: MOCK_EXEC_TX_CALL_DATA,
+        };
+
+        const result = await request(app.getHttpServer())
+          .post('/v1/relay')
+          .send(body)
+          .expect(201);
+
+        expect(result.body).toStrictEqual({ taskId: '123' });
+
+        expect(sponsorServiceSpy).toHaveBeenCalledTimes(1);
```

Comment:
Personal opinion: we started doing this initially in the CGW, however it can become quite overwhelming as the service evolves. As the amount of calls to different parts grow and the number of parameters change we thought that we were not getting a lot of value with this.

We see these tests mostly from a user perspective.

Reply:
I've removed the excessive ones but kept those necessary, e.g. for rate limiting tests.